docker/Dockerfile

```diff
@@ -0,0 +1,72 @@
+FROM ubuntu:20.04
+FROM python:3.8 
+
+# Set default UID and GID
+ENV USER_ID 1000
+ENV GROUP_ID 1000
+
+# Set an encoding to make things work smoothly.
+ENV LANG en_US.UTF-8
+
+# create user and install latest miniconda
+COPY docker/requirements.txt /tmp/requirements.txt
+COPY docker/requirements.txt.forge /tmp/requirements_forge.txt
+RUN set -ex \
+ && groupadd -r conda -g $GROUP_ID \
+ && useradd -u $USER_ID -r -g conda -d /home/conda -s /bin/bash -m conda \
+ && chmod 755 /home/conda \
+ && echo 'conda ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers \
+#  && yum update -y \
+#  && yum install -y \
+#       wget \
+#       curl \
+#       sudo \
+#       bzip2 \
+#       zip \
+#       yum-utils \
+#       centos-release-scl-rh \
+# && yum update -y \
+# && yum clean all \
+ && curl -sSL https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -o miniconda.sh \
+ && bash miniconda.sh -b -p /opt/conda \
+ && rm miniconda.sh \
+ && touch /opt/conda/conda-meta/pinned \
+ && export PATH=/opt/conda/bin:$PATH \
+ && conda config --set show_channel_urls True \
+ && conda update --all --yes \
+ && conda install --yes --file /tmp/requirements.txt \
+ && conda install --yes --channel conda-forge --file /tmp/requirements_forge.txt \
+ && conda clean -tipsy \
+ && rm -rf /opt/conda/pkgs \
+# && rm -rf /var/cache/yum \
+ && rm /tmp/requirements.txt \
+ && rm /tmp/requirements_forge.txt
+
+# set default user and workdir
+WORKDIR /home/conda
+ADD dist/DSWx-HLS-0.1.tar.gz .
+
+# Activate the `conda` environment `root` and the devtoolset compiler.
+# Provide a default command (`bash`), which will start if the user doesn't specify one.
+# ENTRYPOINT [ "/opt/docker/bin/entrypoint" ]
+
+USER root
+RUN mkdir -p DSWx-HLS-0.1/build
+RUN export PATH=/opt/conda/bin:$PATH && cd DSWx-HLS-0.1 && python3 setup.py install
+
+RUN mkdir -p runconfig && chown conda:conda runconfig
+RUN mkdir -p input_dir && chown conda:conda input_dir
+RUN mkdir -p output_dir && chown conda:conda output_dir
+RUN mkdir -p scratch_dir && chown conda:conda scratch_dir
```

Comment:
@gshiroma We may need to rethink this line, as its been the source of some permission issues when running the combined PGE/SAS image.

Typically, when running a Docker image that will be mounting Docker volumes, you specify that the container should run as the current user/group of the host machine (via the `-u $UID:$(id -g)` argument in the `docker run` call). We also provide docker volume mounts to the `runconfig`, `input_dir` and `output_dir` directories within the container, which in effect changes the owner/group from `conda:conda` to `$UID:$(id -g)`. This ensures that permissions for both input and output products are set correctly "on both sides" of the container. 

However, this still leaves `scratch_dir`, which since we don't mount anything to it, it is still owned by `conda:conda`. This has caused permissions errors when the PGE (running as `$UID:$(id -g)`) tries to write out a SAS-only RunConfig to feed to the Python executable. Either the PGE or SAS (or both) should be responsible for creating `scratch_dir` at runtime based on how the RunConfig is set, or the Dockerfile needs to fully open permissions so any user/group running the container can write to it.

Personally I'm for the former option of creating it dynamically, since the PGE already has logic to do this. Let me know y'alls thoughts on this.

Reply:
Thank you, @collinss-jpl . Good point. I updated the directory permissions on commit [b10f426](https://github.com/opera-adt/PROTEUS/pull/14/commits/b10f426bd5658729b37c44efbd38229e1ca3ec33) substituting `conda:conda` with `$UID:$(id -g)` following your suggestion. I'm doing the same for `scratch_dir` which can also be mounted and handled the same way as the other directories. Would that address your concerns?

Reply:
Hi @gshiroma, thanks for taking a look. Unfortunately this approach does not address this issue, as changing ownership to `$UID:$(id -g)` from *within* the Dockerfile would just set the user and owner to `root` (since line 53 of the Dockerfile changes the user to `root` before the directories are created).

When I talk about "opening up" the permissions, I meant something more like this:

`RUN mkdir -p scratch_dir && chown conda:conda scratch_dir && chmod 777 scratch_dir`

So basically the directory is created with fully open permissions for any user/group that runs the container. This should only need to be done for `scratch_dir`, since we mount the other directories as Docker volumes.

Reply:
Thank you again, @collinss-jpl . I accepted your suggestions to open up `scratch_dir` permissions to all groups and users. I also reverted changes to other directories. Please, let me know if that's correct now.

Reply:
Looks great, thanks for the prompt response